torch/nn/functional.py

```diff
@@ -2496,11 +2520,18 @@ def _output_size(dim):
 
         # make scale_factor a tensor in tracing so constant doesn't get baked in
         if torch._C._get_tracing_state():
-            return [(torch.floor((input.size(i + 2).float() * torch.tensor(scale_factors[i],
+            return [(torch.floor((input.size(i + 2).float() * torch.scalar_tensor(scale_factors[i],
```

Comment:
Why was this changed to scalar_tensor? Related: I can't find `torch.scalar_tensor` in docs anywhere.

Reply:
This is actually unrelated. 
Scalar_tensor will create a tensor as well, which doesn't affect PyTorch, but makes a difference on the ONNX side (tensor is saved a constant in ONNX but not scalar_tensor).
If there is any issues with this line, I could do it in a separate PR.

Reply:
yea i guess that would be preferable, mostly just because it's an undocumented function that's not used anywhere else in pytorch library code.